src/components/Tabs/Tabs.tsx

```diff
@@ -0,0 +1,77 @@
+import { KeyboardEvent, ReactElement, useRef } from "react"
+import { TabList } from "./components/TabList"
+import { TabProps } from "./components/Tab"
+import { TabPanel } from "./components/TabPanel"
+import { Tab } from "./types"
+
+interface TabsProps {
+  /**
+   * We constrain `children` to be element(s) with a label and children
+   * props (e.g. `Tab`). This is only structural, however.
+   **/
+  children: ReactElement<TabProps> | Array<ReactElement<TabProps>>
+  currentTab: number
+  onTab: (index: number) => void
```

Comment:
⛏️ nit pick - I prefer onTabChange here, since this event fires off every time the tab selection changes.

Reply:
Ah, except that it actually gets called on every tab click, not just when the tab changes. Well maybe that's something to consider, does it need to be called if the tab is already selected? 🤷

Reply:
@thispastwinter thanks for looking this over! Yeah, I think I tend to agree regarding the naming. I could see `onTabChange` or even just `onChange` making more sense. 👌

Re: Preventing it getting called if already "active"—hmmm, I could see doing this, but I'm trying to weigh the benefits. Do you think it's important?

Reply:
Probably not :)